crates/api/src/lib.rs

```diff
@@ -87,63 +95,73 @@ pub(crate) async fn get_post(post_id: i32, pool: &DbPool) -> Result<Post, LemmyE
   }
 }
 
-pub(crate) async fn get_user_from_jwt(jwt: &str, pool: &DbPool) -> Result<User_, LemmyError> {
+pub(crate) async fn get_local_user_view_from_jwt(
+  jwt: &str,
+  pool: &DbPool,
+) -> Result<LocalUserView, LemmyError> {
   let claims = match Claims::decode(&jwt) {
     Ok(claims) => claims.claims,
     Err(_e) => return Err(ApiError::err("not_logged_in").into()),
   };
-  let user_id = claims.id;
-  let user = blocking(pool, move |conn| User_::read(conn, user_id)).await??;
+  let local_user_id = claims.local_user_id;
+  let local_user_view =
+    blocking(pool, move |conn| LocalUserView::read(conn, local_user_id)).await??;
   // Check for a site ban
-  if user.banned {
+  if local_user_view.person.banned {
     return Err(ApiError::err("site_ban").into());
   }
-  Ok(user)
+  Ok(local_user_view)
 }
```

Comment:
Big change here, this uses the local_user_id, rather than the user_id (which is what the person table moved to)

Reply:
So all existing jwt tokens will be invalidated, and users will be logged out? Did you test if this is handled correctly in the frontend?

Reply:
Correct. Yep I did test it on the front end, and you can test it as well on the federation test instances. Your front end will receive a special error message, and it knows to unset your jwt cookie if it receives that, and tell the user they're logged out.